spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -314,6 +314,15 @@ public boolean isInExpectedState() {
 		return isRunning() || isStoppedNormally();
 	}
 
+	@Override
+	public void enforceRebalance() {
+		super.enforceRebalance();
```

Comment:
Please, see my comment in your test below and consider to use this `this.thisOrParentContainer.enforceRebalanceRequested.set(true)` instead.